### PR TITLE
feat: n.x and n.m.x always use flatten

### PR DESCRIPTION
**Pull request by dead-horse**
This change is [Reviewable](https://reviewable.io/reviews/cnpm/npminstall/200)

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -434,3 +434,5 @@ exports.getGlobalInstallMeta = prefix => {
   }
   return meta;
 };
+
+exports.endsWithX = version => !!version.match(/^\d+\.(x|\d+\.x)$/);
```
Comment (fengmk2) [inline lib/utils.js]: version 会不存在

`version => version && !!version.match(/^\d+\.(x|\d+\.x)$/);`

Comment (dead-horse) [inline lib/utils.js]: 有确保 pkg.version 一定存在的：

```js
  // default install latest version
  if (!pkg.version) {
    pkg.version = '*';
  }
```

Comment (fengmk2) [inline lib/utils.js]: https://travis-ci.org/cnpm/npminstall/jobs/201031197#L3084

Comment (fengmk2) [inline lib/utils.js]: 看来 version 不是字符串

Comment (dead-horse) [inline lib/utils.js]:
```
{ name: 'pedding', version: 1 }
```

测试用例写的数字，囧

Comment (dead-horse) [inline lib/utils.js]: 加个类型判断吧

**Comment by fengmk2** [comment]
2.23.0

**Pull request merged**